### ISSUE TITLE
 [Feature Request] add "Loaded Tabs"

### ISSUE THREAD

**Issue by nepravelniy** [Loirooriol/tab-counter-plus#26]
Can you please add another counter to show how many tabs was loaded?
I can see two ways of showing this. 

1) add number in brackets 
This window: 6 (1)
All windows: 196 (10)
so it represent how many tabs loaded in this window (1) and in all windows (10)

2) add separate row and display just how many we got across the board
This window: 6 
All windows: 196 
Loaded Tabs: 10

**Comment by Loirooriol** [comment]
I am thinking about a detailed view that could look somewhat like this:

```
┏━━━┳━━━┯━━━┯━━━┯━━━┓
┃ = ┃ … │ 👁️̷ │ 📌 │ 💤 ┃ <─ total, normal, hidden, pinned, discarded
┣━━━╋━━━┿━━━┿━━━┿━━━┫
┃ 6 ┃ 3 │ 2 │ 1 │ 0 ┃ <─ current window
┣━━━╋━━━┿━━━┿━━━┿━━━┫ ╮
┃ 1 ┃ 1 │ 0 │ 0 │ 0 ┃ │
┠───╂───┼───┼───┼───┨ │
┃ 4 ┃ 2 │ 1 │ 0 │ 1 ┃ ├─ other windows
┠───╂───┼───┼───┼───┨ │
┃ 3 ┃ 1 │ 0 │ 1 │ 1 ┃ │
┗━━━┻━━━┻━━━┻━━━┻━━━┛ ╯
```

Thoughts?

**Comment by nepravelniy** [comment]
so there will be 1+x rows, where x is amount of open windows?
im not sure how useful it would be, i personally interested in current window and overall info. it would be way less messy and easier to identify things. 

first i don't see total stats across the browser, which current version does show
secondly your 4th row says ┃ 4 ┃ 2 │ 1 │ 0 │ 1 ┃ but it is kinda hard to identify which exact window it is, so i have cycle between them to fund that sucker and that feel inconvenient. while if you add window title to that row it would become gigantic i think. 

if you want to add some detailed info you should look at ublock approach 
![Screenshot 2020-09-25 142011 png_ ‎- Paint 3D 2020](https://user-images.githubusercontent.com/70824752/94261969-602df780-ff3b-11ea-80cd-1b39975c6f3b.png)

something like that might work. so you have a small row with icons in top with bages near that icons which shows stats
and you can have some text info below that icons. and you can hit more so see grid like you did with title names. ublock is also opensource so i guess you can look up how it is made and modify that menu

**Comment by Loirooriol** [comment]
Thanks for the feedback!

> first i don't see total stats across the browser

There can also be a row with totals

> it is kinda hard to identify which exact window it is

There can be a button to switch to that window

> while if you add window title to that row it would become gigantic i think

Yes, titles can bee too big. But I could show them as tooltips.

> you should look at ublock approach

I guess I was taking more inspiration from uMatrix rather than uBlock :) I will take this into account

**Comment by nepravelniy** [comment]
well i wasn't using uMatrix and it was recently discounted but as far as i remember they have something in common in gui if you unfold ublock by clicking more on the bottom 😃

**Comment by Loirooriol** [comment]
I'm thinking I will defer a possible redesign of the popup, and for now just add some checkboxes (unchecked by default) in the `about:addons` preferences:

![Preferences](https://user-images.githubusercontent.com/7477678/96374563-bb868a80-1173-11eb-9b1c-58ca6b57c19d.png)

They would affect both the toolbar icon and the popup.

If there are excluded tabs in the current window, the popup can look like

![Popup with excluded tabs in current window](https://user-images.githubusercontent.com/7477678/96374666-38196900-1174-11eb-8ea1-cab4290163ed.png)

If there are excluded tabs in other windows,

![Popup with excluded tabs in other windows](https://user-images.githubusercontent.com/7477678/96374595-eec91980-1173-11eb-9045-aaf84be63378.png)

Or if there is no excluded tab, the current behavior:

![Popup with no excluded tab](https://user-images.githubusercontent.com/7477678/96374596-f2f53700-1173-11eb-83d5-c6fe106d3511.png)

Thoughts?

**Comment by nepravelniy** [comment]
Looks awesome. I like preferences menu. 

but do we really need such a long "excluded" tag? 

and it depends why you need that plugin. personally i use it to force myself clean some messy windows. if it will show me only loaded once i wont be so bothered cleaning them. its not hard to count 25+91 but thats still some effort. shouldn't it show - total 116 ( 96 excluded)? or even opposite way - total 116 (25 loaded)?

**Comment by Loirooriol** [comment]
> but do we really need such a long "excluded" tag?

Yeah kind of long, but otherwise it's not much clear. I'm also considering "excl.", "other" or "more"

> and it depends why you need that plugin

The case that I'm trying to cover is when you are not interested in some kinds of tabs, like unloaded ones since they have a much smaller impact. Then the toolbar icon doesn't exclude them so that you see the number of interesting tabs. And the popup shows the same number for consistency. Then the number of excluded tabs is an extra info.

Then, would it be better to provide the total instead of the exclusions?
```
This window:  6  (+5 excl.)
All windows: 20 (+96 excl.)
```
vs.
```
This window:  6  (11 total)
All windows: 20 (116 total)
```

I'm not sure. I feel like the latter is less clear about why the main number is smaller than the total, while in the former case it's explicit that some tabs have been excluded. But both can be reasonable. I can consider adding a pref to decide.

**Comment by nepravelniy** [comment]
> I can consider adding a pref to decide.
that's a win-win situation here! 

> I'm also considering "excl.", "other" or "more"
personally i like short excl. or even somekind of emoji 💤

**Comment by nepravelniy** [comment]
> The case that I'm trying to cover is when you are not interested in some kinds of tabs, like unloaded ones since they have a much smaller impact. Then the toolbar icon doesn't exclude them so that you see the number of interesting tabs. And the popup shows the same number for consistency. Then the number of excluded tabs is an extra info.

what about adding both? your where mentioning that you want to make some kind of grind with all info. i think both variants are great and would love any way you chose. just to mention it

**Comment by Loirooriol** [comment]
> what about adding both?

Yes, I think there should be a detailed view in which you can get full info. This is just for the basic view, that I want to keep simple.

> or even somekind of emoji 💤.

I think 💤 works well for unloaded tabs, but there can be other kinds of exclude tabs. And not sure if there is some emoji that can represent "excluded" well.

**Comment by cognitiaclaeves** [comment]
Can I jump in here? I'm hoping to replace the tab tally add-on:
<img width="188" alt="image" src="https://user-images.githubusercontent.com/8933161/119714794-e1ba8a00-be28-11eb-9dcf-4bc5f9ed237f.png">

The reason why I'm using both is because tab tally specifically tells me how many tabs are loaded, which I can use to determine if my awful tabits are causing my user experience to suck. Ideally, I would just close tabs when I'm done using them, but sometimes I'm at my limit of executive function while researching solutions to problems.

So, for the interim, I want to track the total tabs that are loaded, in addition to everything else your plug in currently counts.

I don't know if I care about pinned tabs, or other kinds of excluded tabs -- unless they would be considered not loaded and still actively contribute to resource depletion with the same intensity as loaded tabs. I'm also fine with needing to hit an extra button or key for getting this info -- if the tide of the solution drives more toward not having a separate line item for loaded.

I'd prefer not to have a count where I can't see both total and loaded, though -- so if there's an option that straight-up excludes unloaded tabs from the count, it would hopefully be optional, implemented in a way where I can still see both totals and loaded.
